src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs

```diff
@@ -39,11 +44,25 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
             } 
         }
 
-        private async Task RMarkdownRenderAsync(IRSession session, IFileSystem fs, string inputFilePath, string outputFilePath, string format, int codePage) {
+        private async Task RMarkdownRenderAsync(IRSession session, IFileSystem fs, string inputFilePath, string outputFilePath, string format, int codePage, IVsStatusbar statusBar) {
             using (var fts = new DataTransferSession(session, fs)) {
-                var rmd = await fts.SendFileAsync(inputFilePath);
-                var publishResult = await session.EvaluateAsync<ulong>($"rtvs:::rmarkdown_publish(blob_id = {rmd.Id}, output_format = {format.ToRStringLiteral()}, encoding = 'cp{codePage}')", REvaluationKind.Normal);
-                await fts.FetchFileAsync(new RBlobInfo(publishResult), outputFilePath);
+                string currentStatusText;
+                statusBar.GetText(out currentStatusText);
```

Comment:
Doesn't IVs* use COM STA marshaling everywhere? I recall it was even deliberately done for some services that switched to managed code, so as to not break that contract.